pkg/command/root.go

```diff
@@ -121,6 +101,34 @@ func NewRootCmd() (*cobra.Command, error) {
 	return rootCmd, nil
 }
 
+func newRootCmd() *cobra.Command {
+	var rootCmd = &cobra.Command{
+		Use: "tanzu",
+		// Don't have Cobra print the error message, the CLI will
+		// print it itself in a nicer format.
+		SilenceErrors: true,
+		// silencing usage for now as we are getting double usage from plugins on errors
+		SilenceUsage: true,
+		// Flag parsing must be deactivated because the root plugin won't know about all flags.
+		DisableFlagParsing: true,
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			if cmd.Name() != cobra.ShellCompRequestCmd && cmd.Name() != completionCmd.Name() &&
+				cmd.Name() != newVersionCmd().Name() &&
+				!(cmd.Name() == "set" && (cmd.Parent().Name() == configCmd.Name() || cmd.Parent().Name() == newCEIPParticipationCmd().Name())) {
```

Comment:
nit: It might be more readable if we can put this condition out in its own function.

Reply:
Updated